File: Python developer/step07/7.03/7.3.04.py
'''

Дан список чисел, записанных через пробел. Реализуйте кучу с максимумом в вершине для этого списка.

Далее программе на вход будут поступать команды:

    insert <число> – добавить число в кучу;
    pop – удалить вершину и напечатать ее;
    end – завершить выполнение программы.

Ввод:

    nums – список чисел записанных через пробел
    Список команд.

Вывод:

Результаты работы программы

Sample Input:

1 5 2 6 3 7 4 8
insert 20
pop
insert 0
pop
insert 4
pop
end

Sample Output:

20
8
7

'''

from heapq import heappush, heappop

nums = [int(i)*-1 for i in input().split()]
nums.sort()

while True:
    command = input().split()
    match command[0]:
        case 'insert':
            heappush(nums, int(command[1])*-1)
        case 'pop':
            print(heappop(nums)*-1)
        case 'end':
            exit(0)

File: Fundamentals of C programming/step07/7.3/7.3.04.c
/*

Дубликаты
Удалить из последовательности дубликаты.

Входные данные:
В первой строке задано натуральное число NN. Во второй строке записана упорядоченная по возрастанию последовательность из NN целых чисел, которая может содержать одинаковые элементы.

Выходные данные:
Вывести все различные элементы последовательности, упорядоченные по возрастанию.

Sample Input 1:

21
 -2 -2 -2 4 5 6 7 7 7 7 7 7 7 7 91 91 92 92 93 93 96

Sample Output 1:

-2 4 5 6 7 91 92 93 96

Sample Input 2:

9
 1 2 4 5 6 7 8 9 10

Sample Output 2:

1 2 4 5 6 7 8 9 10

Sample Input 3:

1
 -7

Sample Output 3:

-7

*/

#include <stdio.h>
 
int main() {
int a, b,x=-9999;
scanf("%d", &a);
for (int i=0 ;  i <= (a-1); i++){
scanf("%d", &b);
 if(b > x)
printf("%d ", b );   
x=b;
}
  return 0;
}
